tests/php/modules/protect/test-shared-functions.php

```diff
@@ -0,0 +1,232 @@
+<?php
+/**
+ * Tests for modules/protect/shared_functions.php
+ *
+ * @package Jetpack
+ */
+
+// Require the functions.
+require_once JETPACK__PLUGIN_DIR . '/modules/protect/shared-functions.php';
+
+/**
+ * Tests for modules/protect/shared_functions.php
+ */
+class WP_Test_Jetpack_Protect_Shared_Functions extends WP_UnitTestCase {
+
+	/**
+	 * Test `jetpack_protect_get_ip` and `jetpack_clean_ip`.
+	 *
+	 * @covers jetpack_protect_get_ip
+	 * @covers jetpack_clean_ip
+	 * @dataProvider provide_jetpack_protect_get_ip
+	 * @param string|false $expect Expected output.
+	 * @param array        $server Data for `$_SERVER`.
+	 * @param object|null  $trusted_header_data Trusted header data.
+	 */
+	public function test_jetpack_protect_get_ip( $expect, $server, $trusted_header_data ) {
+		add_filter(
```

Comment:
To avoid leaving things behind once we've completed this series of tests, should we `remove_filter` in a `tearDown` method?

Reply:
It looks like `WP_UnitTestCase_Base::setUp()` and `WP_UnitTestCase_Base::tearDown()` already handle that, they back up and restore the global `$wp_filter` (and `$wp_actions` and `$wp_current_filter`).

I could add a comment here to make that clear, if you'd like.

Reply:
Oh cool, I did not know that. 

> I could add a comment here to make that clear, if you'd like.

No, that's alright. I should have looked before to comment. :)